### PR TITLE
BREAKING: Clean up code and remove deprecations - WIP/Review - 0.9

### PR DESCRIPTION
**Pull request by evs-chris**
**Description of the pull request:**

* [x] Remove old-style event, decorator, and transition syntax
* [x] Adjust directive parsing to use the normal expression parser, so that parsing attributes is a bit more robust.

    For instance, this is now possible: `<button on-click="@this.resetPartial("foo", "the new partial <b>with tags</b> and <span data-foo=attributes>inside</span> an attribute" + someRef)">yep</button>`
* [x] Remove partial comments support (`<!-- {{>somePartial}} --> ... <!-- {{/somePartial}} -->`)
* [x] Remove deprecated event object properties
* [ ] Clean up event context helpers/merge more code with regular prototype methods if possible

After converting the tests for the proxy event with args removal, I'm thinking that a fire helper for replacing `<button on-click="eventName:arg1, arg2">`, which now becomes `<button on-click="@this.fire('eventName', event, arg1, arg2)">`, with `<button on-click="event.fire('eventName', arg1, arg2)">`. Note that depending on what the arguments actually are, the proxy version may come up with a completely weird single or multiple args when it actually fires due to proxy event parsing jankiness. Adding that `fire` to the event object would also be a tiny addition.

**Fixes the following issues:**
#2431 

**Is breaking:**
Yep. The template version will need to be bumped again because the directive args got significantly simplified for most cases.

**Reviewers:**
Yes, please.

### CODE REVIEW

**Comment by fskreuz** [comment]
`+315 −1,080`

😱 some serious housekeeping going on here.

File: src/parse/converters/element/readAttribute.js [hunk 1]
```diff
@@ -197,57 +189,60 @@ export function readAttributeOrDirective ( parser ) {
...
-				parser.error( 'Cannot use reserved event names (change, reset, teardown, update, construct, config, init, render, unrender, detach, insert)' );
+			// check for a proxy event
+			if ( !readProxyEvent( parser, attribute ) ) {
+				// otherwise, it's an expression
+				readArguments( parser, attribute, true );
+			} else {
+				if ( reservedEventNames.test( attribute.f ) ) {
+					parser.pos -= attribute.f.length;
+					parser.error( 'Cannot use reserved event names (change, reset, teardown, update, construct, config, init, render, unrender, detach, insert)' );
+				}
```
Comment (fskreuz) [inline src/parse/converters/element/readAttribute.js]: Collapse to `if-else if`?

```
if ( !readProxyEvent( parser, attribute ) ) {
  ...
} else if ( reservedEventNames.test( attribute.f ) ) {
  ...
}
```

File: src/parse/converters/element/readAttribute.js [hunk 2]
```diff
@@ -197,57 +189,60 @@ export function readAttributeOrDirective ( parser ) {
...
 		}
 
 		else {
+			parser.allowWhitespace();
+			if ( parser.nextChar() === '=' ) {
+				const value = readAttributeValue( parser );
+				if ( value != null ) { // not null/undefined
+					attribute.f = value;
+				}
+			}
```
Comment (fskreuz) [inline src/parse/converters/element/readAttribute.js]:
```
let value = parser.nextChar() === '=' ? readAttributeValue( parser ) : null;
attribute.f = value !== null ? value : attribute.f;
```

File: src/parse/converters/element/readAttribute.js (hunk 1 above)
Comment (evs-chris) [inline src/parse/converters/element/readAttribute.js]: Derp :smile:

**Pull request merged**